src/category_theory/is_connected.lean

```diff
@@ -56,39 +66,51 @@ This allows us to show that the functor X ⨯ - preserves connected limits.
 
 See https://stacks.math.columbia.edu/tag/002S
 -/
-class connected (J : Type v₂) [category.{v₁} J] extends inhabited J :=
-(iso_constant : Π {α : Type v₂} (F : J ⥤ discrete α), F ≅ (functor.const J).obj (F.obj default))
+class is_connected (J : Type v₂) [category.{v₁} J] extends is_preconnected J : Prop :=
```

Comment:
Why do you chose to add a parameter instead of just requiring that it extends `is_preconnected J` and `nonempty J`? If there is a good reason to do so, can you comment on this in the docstring?